openmeter/ent/schema/customer.go

```diff
@@ -0,0 +1,85 @@
+package schema
+
+import (
+	"entgo.io/ent"
+	"entgo.io/ent/schema/edge"
+	"entgo.io/ent/schema/field"
+	"entgo.io/ent/schema/index"
+
+	"github.com/openmeterio/openmeter/pkg/framework/entutils"
+	"github.com/openmeterio/openmeter/pkg/models"
+)
+
+// Customer stores information about a customer
+type Customer struct {
+	ent.Schema
+}
+
+func (Customer) Mixin() []ent.Mixin {
+	return []ent.Mixin{
+		entutils.ResourceMixin{},
+	}
+}
+
+func (Customer) Fields() []ent.Field {
+	return []ent.Field{
+		field.String("currency").GoType(models.CurrencyCode("")).MinLen(3).MaxLen(3).Optional().Nillable(),
+		field.Enum("tax_provider").GoType(models.TaxProvider("")).Optional().Nillable(),
+		field.Enum("invoicing_provider").GoType(models.InvoicingProvider("")).Optional().Nillable(),
+		field.Enum("payment_provider").GoType(models.PaymentProvider("")).Optional().Nillable(),
+		field.String("external_mapping_stripe_customer_id").Optional().Nillable(),
+
+		// PII fields
+		field.String("primary_email").Optional().Nillable(),
+		field.String("address_country").MinLen(2).MaxLen(2).Optional().Nillable(),
+		field.String("address_postal_code").Optional().Nillable(),
+		field.String("address_state").Optional().Nillable(),
+		field.String("address_city").Optional().Nillable(),
+		field.String("address_line1").Optional().Nillable(),
+		field.String("address_line2").Optional().Nillable(),
+		field.String("address_phone_number").Optional().Nillable(),
+	}
+}
+
+func (Customer) Indexes() []ent.Index {
+	return []ent.Index{}
+}
+
+func (Customer) Edges() []ent.Edge {
+	return []ent.Edge{
+		edge.To("subjects", CustomerSubjects.Type),
+	}
+}
+
+// CustomerSubject stores the subject keys for a customer
+type CustomerSubjects struct {
```

Comment:
We might be better off by using a char slice instead of this table, or at least I would not use soft-delete here (e.g. there should not be a deleted_at field).